mycore-ocfl/src/main/java/org/mycore/ocfl/metadata/MCROCFLXMLMetadataManager.java

```diff
@@ -317,7 +328,34 @@ public IntStream getStoredIDs(String project, String type) throws MCRPersistence
 
     @Override
     public int getHighestStoredID(String project, String type) {
-        return getStoredIDs(project, type).max().orElse(0);
+        Path path
+            = Path.of(MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository." + repositoryKey + ".RepositoryRoot"));
+        int highestStoredID = 0;
+        int maxDepth = Integer.MAX_VALUE;
+        OcflExtensionConfig config = MCRConfiguration2.getSingleInstanceOf("MCR.OCFL.Repository." + repositoryKey)
+            .map(MCROCFLRepositoryProvider.class::cast).orElseThrow().getExtensionConfig();
+        // optimization for known layouts
+        if (Objects.equals(config.getExtensionName(), MCRStorageLayoutExtension.EXTENSION_NAME)) {
+            maxDepth = ((MCRStorageLayoutConfig) config).getSlotLayout().split("-").length;
+            path = Path.of(path.toString(), MCROCFLObjectIDPrefixHelper.MCROBJECT.replace(":", ""), project, type);
+        } else if (Objects.equals(config.getExtensionName(),
+            HashedNTupleIdEncapsulationLayoutExtension.EXTENSION_NAME)) {
+            maxDepth = ((HashedNTupleIdEncapsulationLayoutConfig) config).getNumberOfTuples() + 1;
+        }
+        try (Stream<Path> stream
```

Comment:
use of String.startsWith instead of pattern could bring another 1% improvement:
```
Stream<Path> stream
            = Files.find(path, maxDepth, (filePath, fileAttr) -> filePath.getFileName().toString().startsWith(TEST_PREFIX))
````